ecommerce/src/main/kotlin/io/hhplus/ecommerce/product/usecase/ProductStatisticsQueryUseCase.kt

```
@@ -0,0 +1,227 @@
+    private fun getMostViewedProducts(limit: Int): List<Product> {
+        val allProductsWithViews = mutableListOf<Pair<Long, Long>>()
+
+        // TODO: 성능 최적화 필요 - Redis Sorted Set으로 관리 고려
+        // 현재는 샘플링 방식으로 상위 1000개 상품만 체크
+        for (productId in 1L..1000L) {
+            try {
+                val viewCount = eventBasedStatisticsService.getLast10MinuteViews(productId)
+                if (viewCount > 0) {
+                    allProductsWithViews.add(productId to viewCount)
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)